### PR TITLE
Homework.lambda.part1.exercise2 (Mariia_Borovikova)

### PR DESCRIPTION
**Pull request by bormariya**

### CODE REVIEW

File: src/test/java/lambda/part1/exercise/Exercise2.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        testIntegerMultiplier(multiplier);
+    }
+
+    @Test
+    void implementsIntegerMultiplierUsingExpressionLambda() {
+        Multiplier<Integer> multiplier = null;
+
+        multiplier = (i1, i2) -> i1*i2;
+
+        testIntegerMultiplier(multiplier);
```
Comment (Ilya21) [inline src/test/java/lambda/part1/exercise/Exercise2.java]: I think you can inline multiplier

File: src/test/java/lambda/part1/exercise/Exercise2.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        }
+        String result = joiner.toString();
+        return result.equals(delimiter) ? "" : result;
+    }
+
+    @Test
+    void implementsStringMultiplierUsingObjectMethodReference() {
+        Multiplier<String> multiplier = null;
+
+        multiplier = new Exercise2()::stringSumWithDelimiter;
```
Comment (Ilya21) [inline src/test/java/lambda/part1/exercise/Exercise2.java]: try to use `this::stringSumWithDelimeter`

**Review by Ilya21 — APPROVED** [review]

File: src/test/java/lambda/part1/exercise/Exercise2.java (hunk 1 above)
Comment (bormariya) [reply src/test/java/lambda/part1/exercise/Exercise2.java]: @Ilya21 Yes, it looks better, thanks. Fixed)

File: src/test/java/lambda/part1/exercise/Exercise2.java (hunk 2 above)
Comment (bormariya) [reply src/test/java/lambda/part1/exercise/Exercise2.java]: Aha, you're right, I fixed it)

Comment (bormariya) [reply src/test/java/lambda/part1/exercise/Exercise2.java]: @Ilya21 Aha, you're right, I fixed it)

**Review by Nisuuum — APPROVED** [review]

**Pull request merged**